src/server/pkg/worker/master.go

```diff
@@ -261,14 +265,20 @@ nextInput:
 
 // jobManager feeds datums to jobs
 func (a *APIServer) runJob(ctx context.Context, jobInfo *pps.JobInfo, pool *grpcutil.Pool) error {
```

Comment:
Yup, I agree there's redundant stuff in here. I'll go through and remove those bits. I should've done that before asking for feedback. My bad.

However, I do think there are some net new things in there that are helpful:

- having the timing logged for the downloadInput / runUserCode / uploadInput helpers I think is helpful for us. Yes we could look at logs and look at timestamps ... but wrestling w log interfaces (I'm looking at you GKE) makes this tedious and difficult
- also in my experience having individual timing (for helpers) and a total timing (for the overall call .. in our case `Process()`) is invaluable for detecting bugs introduced in other parts of the code / the overhead of the larger call ... I've seen this pattern a lot and like it
- I agree ... having egress logging / timing is helpful
- having the log pattern for an API call started / ended can be very helpful ... this would've saved me a day or so debugging w a current customer (just to realize that the request is being dropped in between the worker `Process()` call ... and the `jobManager` waiting for the response. I would've seen that the single run of jobmanager for this datum ... on this retry ... had a starting log line but not an ending log line)

More broadly ... I agree some of this is mostly just going to help us debugging. I'm not sure why we're concerned about making the logs too big. In cases where I'm debugging now it's usually backed by a logging service. If we're worried about the noise ... then it's time to use log levels. I'm happy to add some of these in on the `debug' level only.

It's important to make it easier for us to debug. I think we've all discussed how we want to decrease turn around time for customer engagements (pocs, issues, or otherwise). Better debug tools for us makes that possible.

I clearly need to do another pass at this PR before it's ready for review. I'll ping you again when it's ready. But I want to hear your take on the above.